# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-am-build libxmu-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.68; am-1.11)
aclocal; autoheader; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install src/twm $DESTDIR/x11/bin
install -d $DESTDIR/x11/share/X11/twm
install -m 644 src/system.twmrc $DESTDIR/x11/share/X11/twm

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/twm-build.pc << EOF
Name: twm
Version: 1.0.5
Description: Tab Window Manager for the X Window System
Requires: $REQUIRES

exec=\\
/x11/bin/twm \\
/x11/share/X11/twm/system.twmrc
EOF

exit

# patch
--- twm-1.0.5.orig/src/system.twmrc
+++ twm-1.0.5/src/system.twmrc
@@ -8,6 +8,7 @@
 # any windows and to use meta-keys.
 #
 
+RandomPlacement
 NoGrabServer
 RestartPreviousState
 DecorateTransients

# source
https://x.org/releases/individual/app/twm-1.0.5.tar.gz
